extensions/exchanges/gdax/exchange.js

```diff
@@ -1,20 +1,57 @@
 var Gdax = require('gdax')
-  , path = require('path')
-  , colors = require('colors')
-  , numbro = require('numbro')
 
 module.exports = function container (get, set, clear) {
   var c = get('conf')
 
-  var public_client = {}, authed_client
+  var public_client = {}, authed_client, websocket_client = {}, websocket_cache = {}
 
   function publicClient (product_id) {
     if (!public_client[product_id]) {
+      websocketClient(product_id)
       public_client[product_id] = new Gdax.PublicClient(product_id, c.gdax.apiURI)
     }
     return public_client[product_id]
   }
 
+  function websocketClient (product_id) {
+    if (!websocket_client[product_id]) {
+      // OrderbookSync extends WebsocketClient and subscribes to the 'full' channel, so we can use it like one
+      var auth = null
+      if (c.gdax && !c.gdax.key && c.gdax.key !== 'YOUR-API-KEY') {
```

Comment:
Typo here: should be `&& c.dax.key &&` -- right now the logic is ignoring authentication.

I tried doing websocket authentication locally and the exchange rejected my ip, even though I'd put it in the whitelist.  Others' mileage may vary.

Reply:
You're correct!